cpp/cppsdk/gsdk.cpp

```diff
@@ -403,81 +472,144 @@ namespace Microsoft
                 decodeHeartbeatResponse(m_receivedData);
             }
 
-            Microsoft::Azure::Gaming::GSDKInternal& GSDKInternal::get()
+            Microsoft::Azure::Gaming::GSDKInternal* GSDKInternal::get()
             {
-                std::unique_lock<std::mutex> lock(m_gsdkInitMutex);
+                return m_instance;
+            }
+
+            bool GSDK::start(bool debugLogs)
+            {
+                // Return true if we already made the 
+                if (GSDKInternal::m_instance != nullptr)
+                    return true;
+
+                GSDKInternal::m_debug = debugLogs;
 
-                if (!m_instance)
+                auto *instance = new Microsoft::Azure::Gaming::GSDKInternal();
+                if (instance->init())
                 {
-                    m_instance = std::make_unique<GSDKInternal>();
+                    GSDKInternal::m_instance= instance;
                 }
-                return *m_instance;
+                else
+                {
+                    instance->dispose();
+                    delete instance;
+                }
+
+                return GSDKInternal::m_instance != nullptr;
             }
 
-            void GSDK::start(bool debugLogs)
+            void GSDK::stop()
             {
-                GSDKInternal::m_debug = debugLogs;
-                GSDKInternal::get();
+                // This deallocates the GSDKInternal and stops it's internal heartbeat thread
+                if (GSDKInternal::m_instance != nullptr)
+                {
+                    GSDKInternal::m_instance->dispose();
+                    delete GSDKInternal::m_instance;
+                    GSDKInternal::m_instance= nullptr;
+                }
             }
 
             bool GSDK::readyForPlayers()
             {
-                if (GSDKInternal::get().m_heartbeatRequest.m_currentGameState != GameState::Active)
+                auto* gsdk= GSDKInternal::get();
```

Comment:
While I applaud your correct use of auto, it's against our internal code standards. So in order to accept this PR, we'll need you to use GSDKInternal*, and not auto* (I can sympathize with you here, but it's just how it is for now).

Reply:
(Please apply to all uses of auto in this file)